src/timer/mod.rs

```diff
@@ -33,26 +43,58 @@ pub struct TimerResult {
     pub status: ExitStatus,
 }
 
+/// Discard the output of a child process.
+fn discard(output: ChildStdout) {
+    const CHUNK_SIZE: usize = 64 << 10;
+
+    #[cfg(target_os = "linux")]
+    {
+        if let Ok(file) = File::create("/dev/null") {
+            while let Ok(bytes) = splice(
+                output.as_raw_fd(),
+                None,
+                file.as_raw_fd(),
+                None,
+                CHUNK_SIZE,
+                SpliceFFlags::empty(),
+            ) {
+                if bytes == 0 {
+                    break;
+                }
+            }
+        }
```

Comment:
There are two potential error cases here.

* If `File::create("/dev/null")` fails (I have no idea if that can happen?), we would skip the entire `while` loop and not read from `output` at all... potentially blocking the benchmarked program?
* `splice` has been in Linux kernels since 2006. But I'm curious what would happen if we ran this on older (or somhow modified) kernels without a `splice` syscall. Would it fail during startup of hyperfine, since the corresponding wrapper in libc wouldn't be available? Or crash with `ENOSYS`? More importantly: is is something we should worry about, or do you think we can safely ignore it?

Reply:
> * If `File::create("/dev/null")` fails (I have no idea if that can happen?), we would skip the entire `while` loop and not read from `output` at all... potentially blocking the benchmarked program?

If anything fails here we will (silently) drop the `ChildStdout` instance which closes the pipe.  The child will get `EPIPE`/`SIGPIPE` the next time they write to stdout.  (No idea what happens on Windows though.)

Might be a good idea to at least log the error.

> * `splice` has been in Linux kernels since 2006. But I'm curious what would happen if we ran this on older (or somhow modified) kernels without a `splice` syscall. Would it fail during startup of hyperfine, since the corresponding wrapper in libc wouldn't be available? Or crash with `ENOSYS`? More importantly: is is something we should worry about, or do you think we can safely ignore it?

Rust 1.57.0 (the MSRV) has these as the minimum supported versions: https://doc.rust-lang.org/1.57.0/rustc/platform-support.html.  So I don't think we have to worry about anything so old that `splice()` is missing.  If you ran `hyperfine` on a system with too old of a `libc` it would fail to start due to the missing symbol.  If `libc` was new enough but the kernel was old, you'd get `ENOSYS`.

On the other hand, it could be blocked by `seccomp()`, or it could fail for some other reason.  Maybe we should fall back to the `read()`/`write()` loop before we give up.